docs/oauth/overview.md

```diff
@@ -0,0 +1,210 @@
+
+### **Phase 6: Account Recreation**
+
+- **Purpose:** Use the retrieved data to recreate the user’s account on the new service.
+- **Process:**
+    1. The Destination imports the user’s exported data.
+    2. The user resumes activity seamlessly on the new platform.
+    3. This phase concludes the portability process.
+
+## **Security Measures**
+
+- **Authorization Code Flow** – Chosen for confidential clients (most secure option).
+- **Redirect URI Validation** – Prevents open redirect vulnerabilities.
+- **State Parameter** – Mitigates CSRF attacks.
+- **Hashed Client Secrets** – Stored securely in DB; raw secret shown only once.
+- **HTTPS Enforcement** – Required in production environments.
+
+## **Error Handling & Testing Scenarios**
+
+### **Common Error Cases**
+
+- Invalid client_id or redirect_uri mismatch.
+- Expired or invalid authorization code.
+- Invalid or expired access token.
+- Scope mismatch during token validation.
+
+### **Security Tests**
+
+- Verify state parameter prevents CSRF.
+- Confirm HTTPS and redirect validation.
+- Ensure rate limiting for token endpoints.
+
+### **Data Access Validation**
+
+- Verify returned data is JSON‑LD and LOLA-compliant.
+- Ensure tokens allow only scoped access (e.g., activitypub_account_portability).
+
+## **Implementation Notes**
+
+- **One Application per User**: Simplifies testbed logic and mirrors real-world service isolation.
+- **Credential Rotation**: Future enhancement — add secret regeneration endpoint for compromised clients.
+- **Logging**: Record significant events (registration, authorization, token exchange) without exposing secrets.
+- **Compliance**: Aligns with [RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749) and LOLA portability guidelines.
+
+## **Real-Case Example**
+
+**Scenario:** A developer of “MyFediverse” registers their service and imports data:
+
+1. Registers service at testbed → receives client_id and client_secret.
+2. User at MyFediverse clicks “Import from Testbed.”
+3. Redirects to testbed for login and consent.
+4. Authorization code exchanged for token.
+5. MyFediverse retrieves ActivityPub data (profile, posts, followers).
+
+
+## OAuth Implementation Phases
+
+```
+┌──────────────────┐          ┌──────────────────┐          ┌──────────────────┐
+│                  │          │                  │          │                  │
+│  User (Resource  │          │   Source (Auth   │          │   Destination    │
+│      Owner)      │          │     Server)      │          │     (Client)     │
+│                  │          │                  │          │                  │
+└────────┬─────────┘          └────────┬─────────┘          └────────┬─────────┘
+         │                             │                             │
+         │                             │                             │
+         │                             │  (1) Register Application   │
+         │                             │ ←---------------------------┘
```

Comment:
Same comment as above... we could also leave "register application" out of the flow because it's a custom hack